// @sanity/ui/dist/cjs/styles/box/boxStyle.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.responsiveBoxStyle = void 0;
var helpers_1 = require("../helpers");
var BOX_SIZING = {
    content: 'content-box',
    border: 'border-box',
};
var BOX_HEIGHT = {
    stretch: 'stretch',
    fill: '100%',
};
function responsiveBoxStyle() {
    return [
        responsiveBoxSizingStyle,
        responsiveBoxHeightStyle,
        responsiveBoxOverflowStyle,
        responsiveBoxDisplayStyle,
    ];
}
exports.responsiveBoxStyle = responsiveBoxStyle;
function responsiveBoxDisplayStyle(props) {
    var theme = props.theme;
    var media = theme.sanity.media;
    return helpers_1.responsive(media, helpers_1.getResponsiveProp(props.display), function (display) { return ({
        '&:not([hidden])': { display: display },
    }); });
}
function responsiveBoxSizingStyle(props) {
    var theme = props.theme;
    var media = theme.sanity.media;
    return helpers_1.responsive(media, helpers_1.getResponsiveProp(props.sizing), function (sizing) { return ({
        boxSizing: BOX_SIZING[sizing],
    }); });
}
function responsiveBoxHeightStyle(props) {
    var theme = props.theme;
    var media = theme.sanity.media;
    return helpers_1.responsive(media, helpers_1.getResponsiveProp(props.height), function (height) { return ({
        height: BOX_HEIGHT[height],
    }); });
}
function responsiveBoxOverflowStyle(props) {
    var theme = props.theme;
    var media = theme.sanity.media;
    return helpers_1.responsive(media, helpers_1.getResponsiveProp(props.overflow), function (overflow) { return ({
        overflow: overflow,
    }); });
}
